File: KiteWeb/src/api/menu.ts
import { http } from "@/utils/http";

// 后台API返回的统一结果格式
export type ApiResult<T = any> = {
  success: boolean;
  message: string;
  data: T;
  code: number;
  timestamp: string;
};

// 菜单类型枚举
export enum MenuType {
  /** 目录 */
  Directory = 1,
  /** 菜单 */
  Menu = 2,
  /** 按钮 */
  Button = 3
}

// 菜单信息
export type MenuData = {
  /** 菜单ID */
  id: number;
  /** 父菜单ID */
  parentId: number;
  /** 菜单名称 */
  menuName: string;
  /** 菜单编码 */
  menuCode: string;
  /** 菜单类型 */
  menuType: MenuType;
  /** 路由路径 */
  path?: string;
  /** 组件路径 */
  component?: string;
  /** 菜单图标 */
  icon?: string;
  /** 排序号 */
  sort: number;
  /** 是否显示 */
  isVisible: boolean;
  /** 是否缓存 */
  isCache: boolean;
  /** 是否外链 */
  isFrame: boolean;
  /** 状态 */
  status: number;
  /** 权限标识 */
  permissions?: string;
  /** 备注 */
  remark?: string;
  /** 创建时间 */
  createTime: string;
  /** 更新时间 */
  updateTime: string;
  /** 子菜单列表 */
  children: MenuData[];
};

// 分页结果
export type PagedResult<T> = {
  /** 数据列表 */
  items: T[];
  /** 总数量 */
  totalCount: number;
  /** 页码 */
  pageIndex: number;
  /** 页大小 */
  pageSize: number;
};

// 获取菜单列表请求参数
export type GetMenusRequest = {
  /** 页码 */
  pageIndex?: number;
  /** 页大小 */
  pageSize?: number;
  /** 菜单名称 */
  menuName?: string;
  /** 菜单编码 */
  menuCode?: string;
  /** 菜单类型 */
  menuType?: MenuType;
  /** 状态 */
  status?: number;
};

// 创建菜单请求参数
export type CreateMenuRequest = {
  /** 父菜单ID */
  parentId: number;
  /** 菜单名称 */
  menuName: string;
  /** 菜单编码 */
  menuCode: string;
  /** 菜单类型 */
  menuType: MenuType;
  /** 路由路径 */
  path?: string;
  /** 组件路径 */
  component?: string;
  /** 菜单图标 */
  icon?: string;
  /** 排序号 */
  sort: number;
  /** 是否显示 */
  isVisible: boolean;
  /** 是否缓存 */
  isCache: boolean;
  /** 是否外链 */
  isFrame: boolean;
  /** 状态 */
  status: number;
  /** 权限标识 */
  permissions?: string;
  /** 备注 */
  remark?: string;
};

// 更新菜单请求参数
export type UpdateMenuRequest = {
  /** 父菜单ID */
  parentId: number;
  /** 菜单名称 */
  menuName: string;
  /** 菜单编码 */
  menuCode: string;
  /** 菜单类型 */
  menuType: MenuType;
  /** 路由路径 */
  path?: string;
  /** 组件路径 */
  component?: string;
  /** 菜单图标 */
  icon?: string;
  /** 排序号 */
  sort: number;
  /** 是否显示 */
  isVisible: boolean;
  /** 是否缓存 */
  isCache: boolean;
  /** 是否外链 */
  isFrame: boolean;
  /** 状态 */
  status: number;
  /** 权限标识 */
  permissions?: string;
  /** 备注 */
  remark?: string;
};

// API 返回类型定义
export type MenusResult = ApiResult<PagedResult<MenuData>>;
export type MenuResult = ApiResult<MenuData>;
export type MenuTreeResult = ApiResult<MenuData[]>;
export type CreateMenuResult = ApiResult<number>;
export type UpdateMenuResult = ApiResult<boolean>;
export type DeleteMenuResult = ApiResult<boolean>;

/** 获取菜单列表 */
export const getMenus = (params: GetMenusRequest) => {
  return http.request<MenusResult>("get", "/menu", { params });
};

/** 根据ID获取菜单 */
export const getMenuById = (id: number) => {
  return http.request<MenuResult>("get", `/menu/${id}`);
};

/** 获取菜单树 */
export const getMenuTree = () => {
  return http.request<MenuTreeResult>("get", "/menu/tree");
};

/** 获取用户菜单树 */
export const getUserMenuTree = (userId: number) => {
  return http.request<MenuTreeResult>("get", `/menu/user/${userId}`);
};

/** 创建菜单 */
export const createMenu = (data: CreateMenuRequest) => {
  return http.request<CreateMenuResult>("post", "/menu", { data });
};

/** 更新菜单 */
export const updateMenu = (id: number, data: UpdateMenuRequest) => {
  return http.request<UpdateMenuResult>("put", `/menu/${id}`, { data });
};

/** 删除菜单 */
export const deleteMenu = (id: number) => {
  return http.request<DeleteMenuResult>("delete", `/menu/${id}`);
};

File: KiteServer/src/Domain/Entities/Menu.cs
namespace Domain.Entities;

/// <summary>
/// 菜单实体
/// </summary>
[SugarTable("sys_menu")]
public class Menu : BaseEntity
{
    /// <summary>
    /// 父菜单ID（0表示根菜单）
    /// </summary>
    [SugarColumn(ColumnDescription = "父菜单ID")]
    public long ParentId { get; set; } = 0;

    /// <summary>
    /// 菜单名称
    /// </summary>
    [SugarColumn(ColumnDescription = "菜单名称", Length = 50, IsNullable = false)]
    public string MenuName { get; set; } = string.Empty;

    /// <summary>
    /// 菜单编码（唯一标识）
    /// </summary>
    [SugarColumn(ColumnDescription = "菜单编码", Length = 100, IsNullable = false)]
    public string MenuCode { get; set; } = string.Empty;

    /// <summary>
    /// 菜单类型（1：目录，2：菜单，3：按钮）
    /// </summary>
    [SugarColumn(ColumnDescription = "菜单类型")]
    public Shared.Enums.MenuType MenuType { get; set; } = Shared.Enums.MenuType.Menu;

    /// <summary>
    /// 路由路径
    /// </summary>
    [SugarColumn(ColumnDescription = "路由路径", Length = 200, IsNullable = true)]
    public string? Path { get; set; }

    /// <summary>
    /// 组件路径
    /// </summary>
    [SugarColumn(ColumnDescription = "组件路径", Length = 200, IsNullable = true)]
    public string? Component { get; set; }

    /// <summary>
    /// 菜单图标
    /// </summary>
    [SugarColumn(ColumnDescription = "菜单图标", Length = 100, IsNullable = true)]
    public string? Icon { get; set; }

    /// <summary>
    /// 排序号
    /// </summary>
    [SugarColumn(ColumnDescription = "排序号")]
    public int Sort { get; set; } = 0;

    /// <summary>
    /// 是否显示（0：隐藏，1：显示）
    /// </summary>
    [SugarColumn(ColumnDescription = "是否显示")]
    public bool IsVisible { get; set; } = true;

    /// <summary>
    /// 是否缓存（0：不缓存，1：缓存）
    /// </summary>
    [SugarColumn(ColumnDescription = "是否缓存")]
    public bool IsCache { get; set; } = false;

    /// <summary>
    /// 是否外链（0：否，1：是）
    /// </summary>
    [SugarColumn(ColumnDescription = "是否外链")]
    public bool IsFrame { get; set; } = false;

    /// <summary>
    /// 状态（0：禁用，1：启用）
    /// </summary>
    [SugarColumn(ColumnDescription = "状态")]
    public int Status { get; set; } = 1;

    /// <summary>
    /// 权限标识（多个用逗号分隔）
    /// </summary>
    [SugarColumn(ColumnDescription = "权限标识", Length = 500, IsNullable = true)]
    public string? Permissions { get; set; }

    /// <summary>
    /// 备注
    /// </summary>
    [SugarColumn(ColumnDescription = "备注", Length = 500, IsNullable = true)]
    public string? Remark { get; set; }

    /// <summary>
    /// 子菜单列表（导航属性，不映射到数据库）
    /// </summary>
    [SugarColumn(IsIgnore = true)]
    public List<Menu> Children { get; set; } = new List<Menu>();

    /// <summary>
    /// 父菜单（导航属性，不映射到数据库）
    /// </summary>
    [SugarColumn(IsIgnore = true)]
    public Menu? Parent { get; set; }
}
